app/models/api_response.rb

```diff
@@ -0,0 +1,29 @@
+class ApiResponse < ActiveRecord::Base
+  # Associations ---------------------------------------------------------------
+  has_one :destination
+  has_one :origin
+  has_many :packages
+
+  # Validations ----------------------------------------------------------------
+  CARRIER_REGEX = /\A(UPS)|(USPS)\z/
+  validates :carrier, :total_price, :service_type, presence: true
+  validates_format_of :carrier, with: CARRIER_REGEX
+
+  # Instance Methods -----------------------------------------------------------
+  def log!(shipping_details) # OPTIMIZE: DRY this up with accept_nested_attributes_for?
```

Comment:
Yeah it looks like accepts_nested_attributes would make sense. Great job getting it working first though!